src/read/elf/file.rs

```diff
@@ -104,6 +104,23 @@ where
             })
     }
 
+    fn notes(&self) -> impl Iterator<Item = read::Result<NoteIterator<'data, Elf>>> + '_ {
+        let Self {
+            endian,
+            data,
+            sections,
+            segments,
+            ..
+        } = self;
+        let sections = sections
+            .iter()
+            .filter_map(move |header| header.notes(*endian, *data).transpose());
+        let segments = segments
+            .iter()
+            .filter_map(move |header| header.notes(*endian, *data).transpose());
+        std::iter::empty().chain(sections).chain(segments)
```

Comment:
This now iterates over both sections and segments. This is changing the behaviour, and doesn't belong in a refactoring commit. The previous behaviour was intentional, as stated in the existing comment. For the purpose of build ids it probably doesn't matter, but if this ever becomes a general purpose function then you'll start seeing duplicate notes.

I don't think it's worth trying to do this sort of refactor while keeping the previous behaviour.

Reply:
I restored the previous behavior, but I had to add a dependency on `either` to do so. If this still displeases you, I'll just remove this commit.

Reply:
Yes, it's not worth adding a dependency for this, so please remove it.

Reply:
Sorry, I did leave a comment here yesterday, but then accidentally deleted it.

Please remove this change completely. I still disagree with the style of your last change here, and I don't want to spend more time reviewing it.